### PR TITLE
SQL: Fix bug regarding alias fields with dots

### LINKED ISSUE #37224: SQL: alias type fields not handled properly
Elasticsearch offers support for a type of fields known as `aliases`: https://www.elastic.co/guide/en/elasticsearch/reference/6.x/alias.html

ES-SQL relies on [`_field_caps` ](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-field-caps.html) to build the list of fields for an index and, at the moment, this API doesn't have support for "alias" fields in the sense of differentiating between an alias field and a normal field. In general field aliases are supported in ES-SQL (and are regarded in the same way `_field_caps` regards them - normal fields), but there are some scenarios where these don't work.

For the following mapping:

```
{
  "mappings": {
    "doc": {
      "properties": {
        "foo_alias": {
          "type": "alias",
          "path": "parent.foo"
        },
        "parent": {
          "properties": {
            "foo": {
              "type": "keyword",
              "ignore_above": 1024
            }
          }
        },
        "nested.foo_alias": {
          "type": "alias",
          "path": "parent.foo"
        },
        "nested": {
          "properties": {
            "foo_alias": {
              "type": "alias",
              "path": "parent.foo"
            }
          }
        }
      }
    }
  }
}
```

Both `nested.foo_alias` and `nested`->`foo_alias` alias fields will fail in ES-SQL. The reason is that when index resolution happens, when SQL tries to add `foo_alias` to the `nested` field, this doesn't exist and an exception will be thrown 

```
Server error [Server encountered an error [Cannot find field nested; this is likely a bug]. [SqlIllegalArgumentException[Cannot find field nested; this is likely a bug]
        at org.elasticsearch.xpack.sql.analysis.index.IndexResolver.createField(IndexResolver.java:355)
        at org.elasticsearch.xpack.sql.analysis.index.IndexResolver.mergedMapping(IndexResolver.java:329)
        at org.elasticsearch.xpack.sql.analysis.index.IndexResolver.lambda$resolveAsMergedMapping$3(IndexResolver.java:246)
```

`_field_caps` will report `nested.foo_alias` field but it won't report any `nested` field:

```
        "nested.foo_alias": {
            "keyword": {
                "type": "keyword",
                "searchable": true,
                "aggregatable": true
            }
        }
```

but SQL is expecting a `nested` separate field report in `_field_caps` as well.

Related issue https://github.com/elastic/elasticsearch/issues/37141.

### PR DESCRIPTION
**Pull request by costin**
Field of types aliases with dots in name are returned without a
 hierarchy by field_caps, as oppose to the mapping api or field with
 concrete types.

Fix #37224

### CODE REVIEW

**Comment by costin** [comment]
FTR, see my comments on the background behind the approach taken at:
https://github.com/elastic/elasticsearch/issues/37141#issuecomment-452762304

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java [hunk 1]
```diff
@@ -368,8 +373,13 @@ private static EsField createField(String fieldName, Map<String, Map<String, Fie
...
         return esField;
     }
-    
+
     private static EsField createField(String fieldName, String typeName, Map<String, EsField> props, boolean isAggregateable) {
+        return createField(fieldName, typeName, props, isAggregateable, false);
+    }
+    
+    private static EsField createField(String fieldName, String typeName, Map<String, EsField> props, boolean isAggregateable,
+            boolean isSynthetic) {
```
Comment (astefan) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolver.java]: It seems you are not doing anything with `isSynthetic`? Leftover from a previous attempt?

**Review by astefan — APPROVED** [review]
LGTM. Left one comment.
Also, I would have liked to see more IT tests. Do you think it's worth changing our `test_emp` data source? This can open the door to more complex tests involving alias fields in aggregations and filtering...  one thought keeps popping up in my head regarding [this section](https://www.elastic.co/guide/en/elasticsearch/reference/6.6/alias.html#unsupported-apis) of our documentation and am wondering if we will hit any of these restrictions (especially related to `terms` query) in obscure ways.

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java [hunk 2]
```diff
@@ -97,7 +102,28 @@ public void testMisspelledColumnWithWildcard() {
     public void testColumnWithNoSubFields() {
         assertEquals("1:8: Cannot determine columns for [text.*]", error("SELECT text.* FROM test"));
     }
-    
+
+    public void testFieldAliasTypeWithoutHierarchy() {
+        Map<String, EsField> mapping = new LinkedHashMap<>();
+
+        mapping.put("fld", new EsField("fld", DataType.KEYWORD, emptyMap(), true));
```
Comment (matriv) [inline x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java]: Do you need that in the test?

**Review by matriv — APPROVED** [review]
LGTM! Left one comment

File: x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java (hunk 2 above)
Comment (costin) [reply x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java]: Not anymore, it was used in the previous commit when the alias field was hidden.

**Comment by costin** [comment]
@astefan I think `alias`es are quite young and they'll evolve. From the SQL perspective replacing the alias will give us ultimate flexibility (especially when it comes to source extraction).
However it's not entirely up to us though.

I'd prefer these particular types inside `test_emp_copy` since I don't see us running the same query against it - we can always add a query against the separate index, which is what I've added to this PR.

**Pull request merged**